flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/nodes/dataset/DataSetSort.scala

```diff
@@ -71,11 +78,57 @@ class DataSetSort(
       partitionedDs = partitionedDs.sortPartition(fieldCollation._1, fieldCollation._2)
     }
 
+    val offsetAndFetchDS = if (offset != null) {
+      val offsetIndex = RexLiteral.intValue(offset)
+      val fetchIndex = if (fetch != null) {
+        RexLiteral.intValue(fetch) + offsetIndex
+      } else {
+        Int.MaxValue
+      }
+      if (currentParallelism != 1) {
```

Comment:
Can you swap the branches? In the parallelism check above we first checked for `==1` otherwise it might be confusing.